python/dolfinx/io/utils.py

```diff
@@ -172,7 +176,8 @@ def read_mesh(self, ghost_mode=GhostMode.shared_facet, name="mesh", xpath="/Xdmf
         return Mesh(msh, domain)
 
     def read_meshtags(self, mesh, name, xpath="/Xdmf/Domain"):
-        return super().read_meshtags(mesh._cpp_object, name, xpath)
+        mt = super().read_meshtags(mesh._cpp_object, name, xpath)
+        return MeshTags(mt, None)
```

Comment:
Shouldn't this be 
```suggestion
        return MeshTags(mt, mesh)
```
to get the `ufl_domain` information passed to the `MeshTags` object?